spyder/plugins/editor/lsp/manager.py

```diff
@@ -301,14 +301,16 @@ def update_python_config(self):
             'follow_imports': self.get_option('jedi_definition/follow_imports')
         }
 
+        # Advanced
+        advanced_external = self.get_option('advanced/external')
+
         # Setup options in json
         self.python_config['cmd'] = cmd
         if host in self.LOCALHOST:
             self.python_config['args'] = '--host {host} --port {port} --tcp'
-            self.python_config['external'] = False
         else:
             self.python_config['args'] = ''
-            self.python_config['external'] = True
+        self.python_config['external'] = advanced_external
```

Comment:
This line alone is the actual fix?

Reply:
Yep, in this way we can handle an pyls instant launched outside Spyder (even if it's launched in the same machine where Spyder is running and besides the error in the restart of the client)